src/mesh/mesh.cpp

```diff
@@ -1065,22 +1065,26 @@ void Mesh::DoStaticRefinement(ParameterInput *pin) {
   for (auto dir : {X1DIR, X2DIR, X3DIR})
     nrbx[dir - 1] = mesh_size.nx(dir) / base_block_size.nx(dir);
 
-  auto GetLegacyMeshCoordinate = [this, nrbx](CoordinateDirection dir, BlockLocation bloc,
-                                              const LogicalLocation &loc) -> Real {
-    auto xll = loc.LLCoord(dir, bloc);
-    auto root_fac = static_cast<Real>(1 << GetLegacyTreeRootLevel()) /
-                    static_cast<Real>(nrbx[dir - 1]);
-    xll *= root_fac;
-    return this->mesh_size.xmin(dir) * (1.0 - xll) + this->mesh_size.xmax(dir) * xll;
-  };
-
-  auto GetLegacyLLFromMeshCoordinate = [this, nrbx](CoordinateDirection dir, int level,
-                                                    Real xmesh) -> std::int64_t {
-    auto root_fac = static_cast<Real>(1 << GetLegacyTreeRootLevel()) /
-                    static_cast<Real>(nrbx[dir - 1]);
-    auto xLL = (xmesh - this->mesh_size.xmin(dir)) /
-               (this->mesh_size.xmax(dir) - this->mesh_size.xmin(dir)) / root_fac;
-    return static_cast<std::int64_t>((1 << std::max(level, 0)) * xLL);
+  auto GetStaticRefLLIndexRange = [](CoordinateDirection dir, int num_root_block,
+                                     int ref_level, const RegionSize &ref_size,
+                                     const RegionSize &mesh_size) {
+    int lxtot = num_root_block * (1 << ref_level);
+    int lxmin, lxmax;
+    for (lxmin = 0; lxmin < lxtot; lxmin++) {
+      Real r = LogicalLocation::IndexToSymmetrizedCoordinate(lxmin + 1,
+                                                             BlockLocation::Left, lxtot);
+      if (mesh_size.SymmetrizedLogicalToActualPosition(r, dir) > ref_size.xmin(dir))
+        break;
+    }
+    for (lxmax = lxmin; lxmax < lxtot; lxmax++) {
+      Real r = LogicalLocation::IndexToSymmetrizedCoordinate(lxmax + 1,
+                                                             BlockLocation::Left, lxtot);
```

Comment:
Just double checking, this is not meant to be `BlockLocation::Right` is it? (the new logical location machinery is still a little opaque to me).

Reply:
It is supposed to be `BlockLocation::Left` here. I agree this is a little confusing, but this gives code that is as close as possible to the old implementation.